h2o-test-support/src/main/java/water/test/WebsocketClient.java

```diff
@@ -0,0 +1,72 @@
+package water.test;
+
+import com.google.gson.Gson;
+import com.google.gson.reflect.TypeToken;
+import org.glassfish.tyrus.client.ClientManager;
+import water.H2O;
+
+import javax.websocket.*;
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.Map;
+
+import static org.junit.Assert.fail;
+
+public class WebsocketClient extends Endpoint {
+
+    private Map<String, String> receivedMessage;
+    private Session sess;
+
+    private final Gson gson = new Gson();
+    
+    public WebsocketClient() throws URISyntaxException, IOException, DeploymentException {
+        String destUri = "ws://" + H2O.getIpPortString() + "/3/Steam.websocket";
+        ClientManager client = ClientManager.createClient();
+        ClientEndpointConfig cec = ClientEndpointConfig.Builder.create().build();
+        sess = client.connectToServer(this, cec, new URI(destUri));
+    }
+    
+    public void close() throws IOException {
+        sess.close(new CloseReason(CloseReason.CloseCodes.NORMAL_CLOSURE, "Test Done"));
+    }
+
+    @OnOpen
```

Comment:
This annotation is inappropriate here. There is nothing to process it. I'd suggest putting `@Override` there instead.

Reply:
the annotation is processed by Tyrus, this method does not override anything, its called based on the annotation